index.js

```diff
@@ -0,0 +1,153 @@
+require("react-native-get-random-values");
+
+const React = require("react");
+const { StyleSheet, View, Platform } = require("react-native");
+const { WebView } = require("react-native-webview");
+const { MainWorker, webViewWorkerString } = require("webview-crypto");
+
+const styles = StyleSheet.create({
+  hide: {
+    display: "none",
+    position: "absolute",
+
+    width: 0,
+    height: 0,
+
+    flexGrow: 0,
+    flexShrink: 1
+  }
+});
+
+const internalLibrary = `
+(function () {
+  function postMessage (message) {
+    if (window.ReactNativeWebView.postMessage === undefined) {
+      setTimeout(postMessage, 200, message)
+    } else {
+      window.ReactNativeWebView.postMessage(message)
+    }
+  }
+  var wvw = new WebViewWorker(postMessage)
+  //for android
+  window.document.addEventListener('message', (e) => {wvw.onMainMessage(e.data);})
+  //for ios
+  window.addEventListener('message', (e) => {wvw.onMainMessage(e.data);})
+}())
+`;
+
+let resolveWorker;
+let workerPromise = new Promise(resolve => {
+  resolveWorker = resolve;
+});
+
+function sendToWorker(message) {
+  workerPromise.then(worker => worker.onWebViewMessage(message));
+}
+
+const subtle = {
```

Comment:
I see why you needed to add this wrapper, but it's unfortunate it requires hardcoding in all the method names. We already have some of that logic here: https://github.com/saulshanabrook/webview-crypto/blob/master/src/MainWorker.ts

So couple ideas:

1. Update webview-crypto to take a promise to a callback instead of a callback, so that you can instantiate the `MainWorker` at the top level, and then just set the promise in the component
2. Or, don't change webview-crypto, but instantiate the `MainWorker` at the top level still, and for it's callback build up a queue of messages until the component is created.

I think 1 is preferable, but it does require making a PR to the other library first.

Reply:
Actually its not me, Its based on Linus PR.
I didnt question the architecture he did, I just fixed it
maybe its possible just to fix your old code, dono...

Reply:
@saulshanabrook if you want to jump on a video call and explain the architecture and the changes you want, I could probably do it

Reply:
Hey @sirpy that sounds great! What time zone are you in?

Reply:
> Its based on Linus PR

My reasoning for removing TypeScript at the time was to remove as much boilerplate as possible. That's why I switched to `standard` for linting as well. Since the glue script was so small I didn't think that it would give too much ☺️ 

I think that the reason for me never merging that version though was that I took a different approach here: https://github.com/LinusU/react-native-webcrypto

Instead of creating the WebView from the javascript code I create it in native code. The up-side to this is that you just simply `import 'react-native-webcrypto'` and then `crypto.subtle` will be available automagically. You don't have to attach anything to the render tree.

Unfortunately the project a was using it for was put on ice and I didn't get to expand the implementation to iOS or to any more than the specific functions I was using, but that should be easy...

Anyhow, just wanted to give some background, I'm still open to help maintaining this package ☺️

Reply:
@LinusU Oh cool! That sounds like a better approach anyway!

I am also not using this package at all and haven't touched react-native since I last committed to it. I am happy to just add to the README of this package "Go use LinusU's [package](https://github.com/LinusU/react-native-webcrypto) it is much better!"

Reply:
@saulshanabrook @LinusU 
so how do you guys want to go about it?
i can publish this as GoodDollar/react-native-webview-crypto and you can redirect to it in the README or do you want to merge this?

Reply:
@sirpy maybe give LinusU's work a try and see if it works for you as is? If it does, I will add something to the README to just point to that.

Reply:
I'm not sure what you mean. This IS Linus work. I took his PR and fixed it.
@saulshanabrook

Reply:
I mean this repository https://github.com/LinusU/react-native-webcrypto

Reply:
this repo looks like a complete re-write attempt, the IOS part is missing. so its not usable.

Reply:
@saulshanabrook @LinusU 
Do you want to progress with this PR?

Reply:
I just gave you push access to this repo. If you tell me your NPM username, I will give you access to that as well. I am sorry I don't have time to review this! Feel free to do as you feel is best to make progress here. I am happy to still jump on a call if you have any questions about the existing design. @LinusU I also gave you push access as well if you would like to participate!

Reply:
10x. its sirpush on npm

Reply:
I moved all these packages to the same org and invited you to it, since you might need/wanna edit the others as well! Just added you on NPM to them all as well.